docs/guides/tutorials/chatgpt_bot.rst

```diff
@@ -0,0 +1,2316 @@
+
+    // …
+    async function storeEmbeddings() {
+      const client = edgedb.createClient();
+
+      const sectionPaths = await walk("docs");
+
+      console.log(`Discovered ${sectionPaths.length} sections`);
+
+      const sections = await prepareSectionsData(sectionPaths);
+
+      // The rest of the function
+    }
+    // …
+
+We create our EdgeDB client and get our documentation paths by calling
+``walk``. We also log out some debug information showing how many sections were
+discovered. Then, we prep our ``Section`` objects by calling the
+``prepareSectionsData`` function we just walked through and passing in the
+documentation paths we got back from ``walk``.
+
+Next, we'll store this data.
+
+.. code-block:: typescript
+    :caption: generate-embeddings.ts
+
+    // …
+    async function storeEmbeddings() {
+      // The parts we just talked about
+
+      // Delete old data from the DB.
+      await e.delete(e.Section).run(client);
+
+      // Bulk-insert all data into EdgeDB database.
+      const query = e.params({ sections: e.json }, ({ sections }) => {
+        return e.for(e.json_array_unpack(sections), (section) => {
+          return e.insert(e.Section, {
+            content: e.cast(e.str, section.content),
+            tokens: e.cast(e.int16, section.tokens),
+            embedding: e.cast(e.OpenAIEmbedding, section.embedding),
+          });
+        });
+      });
+
+      await query.run(client, { sections });
+      console.log("Embedding generation complete");
+    }
+    // …
+
+The comments do a good job of explaining here, but let's go into a little more
+detail. First, we build and run a query that deletes all ``Section`` objects
+currently in the database. Then, we build another query that will insert the
+new ``Section`` data we just prepared. We await a call to that query's ``run``
+method, passing in the sections we just prepared.
+
+Here's what the whole function looks like:
+
+.. code-block:: typescript
+    :caption: generate-embeddings.ts
+
+    // …
+    async function storeEmbeddings() {
+      const client = edgedb.createClient();
+
+      const sectionPaths = await walk("docs");
+
+      console.log(`Discovered ${sectionPaths.length} sections`);
+
+      const sections = await prepareSectionsData(sectionPaths);
+
+      // Delete old data from the DB.
+      await e.delete(e.Section).run(client);
+
+      // Bulk-insert all data into EdgeDB database.
+      const query = e.params({ sections: e.json }, ({ sections }) => {
+        return e.for(e.json_array_unpack(sections), (section) => {
+          return e.insert(e.Section, {
+            content: e.cast(e.str, section.content),
+            tokens: e.cast(e.int16, section.tokens),
+            embedding: e.cast(e.OpenAIEmbedding, section.embedding),
+          });
+        });
+      });
+
+      await query.run(client, { sections });
+      console.log("Embedding generation complete");
+    }
+    // …
+
+
+Putting it all together
```

Comment:
Do we want this repetition'? Users can go and copy from the example project
I am also fine with leaving this here, just the guide is very long.